src/index.ts

```diff
@@ -12,20 +12,37 @@ type RrwebEvent = {
 export default class SentryRRWeb {
   public readonly name: string = SentryRRWeb.id;
   public static id: string = 'SentryRRWeb';
-  public events: Array<RrwebEvent> = [];
 
-  attachmentUrlFromDsn(dsn: Dsn, eventId: string) {
+  public events: Array<RRWebEvent> = [];
+  private readonly checkoutEveryNms: Number;
+  private readonly checkoutEveryNth: Number;
+
+  public constructor(
+    options: { checkoutEveryNms?: Number; checkoutEveryNth?: Number } = {}
+  ) {
+    // default checkout time of 5 minutes
+    this.checkoutEveryNms = options.checkoutEveryNms || 5 * 60 * 1000;
+    this.checkoutEveryNth = options.checkoutEveryNth;
+  }
+
+  public attachmentUrlFromDsn(dsn: Dsn, eventId: string) {
     const { host, path, projectId, port, protocol, user } = dsn;
     return `${protocol}://${host}${port !== '' ? `:${port}` : ''}${
       path !== '' ? `/${path}` : ''
     }/api/${projectId}/events/${eventId}/attachments/?sentry_key=${user}&sentry_version=7&sentry_client=rrweb`;
   }
 
-  setupOnce() {
+  public setupOnce() {
     rrweb.record({
-      emit(event: RrwebEvent, isCheckout?: boolean) {
+      checkoutEveryNms: this.checkoutEveryNms,
+      checkoutEveryNth: this.checkoutEveryNth,
+      emit(event: RRWebEvent, isCheckout?: boolean) {
         const self = Sentry.getCurrentHub().getIntegration(SentryRRWeb);
```

Comment:
@dcramer My understanding is that `Sentry.getCurrentHub().getIntegration(SentryRRWeb)` could return undefined. I have a check for `if (self) {` in sentry-fullstory for that reason.

Reply:
:+1: